### DISCUSSION TITLE
Advice for H.265 main stream camera?

### DISCUSSION THREAD

**Discussion by bmutari** [blakeblackshear/frigate#1082] [Ask A Question] [answered]
Hi all, I bought a RLC-810A probably prematurely as I've been reading everyone else's experience with it hasn't been the best... I'm looking for some advice on what to do with it. 

I have it working with Frigate and HA just fine using the sub-stream for detection, clips, and rtmp (to HA for live viewing), but obviously that is at a very reduced resolution. I am using the rtsp feed in the Frigate config. Research and trials has shown that the main feed from the RLC-810A is only available in H.265... so I'm kinda sunk for being able to view the recorded clips in HA right? I have tried to lower the resolution on the main feed in hopes of getting it to stream in H.264 but no luck. It's been too long and I cannot return the camera.

Should I shelve it until HA media browser supports H.265 playback? (anyone know anything about that roadmap?) IS there a way I could re-encode the clips to H.264 with ffmepg output options or something like that? If so how resource expensive would that be? (the system is an older i5-2400 with 8GB mem)

Thanks!

**Comment by gpete** [comment]
You might have seen [this issue post](https://github.com/blakeblackshear/frigate/issues/957) already but I wanted to point it out in case you haven't. The author @erikarkenhill uses the 810 and 820s and created a script to do some post processing. I don't know that I've seen the details to his implementation anywhere but if I were to do it with Home Assistant I'd either use the [Node-RED](https://github.com/hassio-addons/addon-node-red/blob/main/node-red/DOCS.md) or [AppDaemon](https://appdaemon.readthedocs.io/en/latest/) listening to MQTT events to trigger a conversion script. Going that route you can use conditionals within Home Assistant to only convert when you want.

I wish I had a better answer because I have an RLC-820A (to my knowledge identical to the 810A other than form factor) but I have side stepped the issue for the moment by relying on my existing Blue Iris setup for reviewing full-resolution clips when I need more detail and having frigate use the sub-stream for clips which are smaller and much easier to retrieve on the go.

**Reply by bmutari** [reply]
hey @gpete, thank you! I had not seen that post about converting the clip after save. Seems like a smarter way to go about that for sure. Might be the best path forward until h.265 is supported... which I hope is soon.

**Comment by digiblur** [comment]
I seem to recall someone saying they tested a new firmware on that camera to fix some Frigate issues. Let me do some digging.

**Reply by bmutari** [reply]
@digiblur !! hey man, love your videos, been a subscriber for a while! I did check firmware on the camera and as far as I can tell it's the most current, dated 1/21/2021 number 21012103, but very interested in what you may have seen before about it.

**Reply by digiblur** [reply]
Thanks!  I have the 810A as well I flashed this file on, I noticed it was for 820 but it worked, I did have to change the option to flip the image over.  This was from another user that had decoding issues with Frigate and such and support him this file that had a new codec in the file supposedly.  https://github.com/digiblur/unraid-docker-templates/tree/master/sample-files/RLC

**Reply by bmutari** [reply]
Great, thanks, will give that a try too.

**Comment by bmutari** [comment]
I think I might be onto something I found this: https://aaron.cc/ffmpeg-hevc-apple-devices/

basically claims that ffmpeg h265 encodes need to be tagged with "hvc1" to allow playback on apple devices.

Right now, I can view the h265 clips from the RLC-810A in the Frigate web UI **AND** in Home Assistant Media Browser in Microsoft Edge (Version 90.0.818.56 (Official build) (64-bit) latest version, however if I'm in Chrome I cannot play the clips. Also have not been able to play the clips in the home assistant app on iOS or via a Safari browser accessing Home Assistant... which I'm now thinking might be because of this tagging of hvc1?? 

Not fully up to speed with how to pass the ffmpeg options in the frigate yaml, any ideas how I might try to get that tag on saved clips?

**Reply by digiblur** [reply]
I read somewhere that only egde and safari support H265?

**Comment by bmutari** [comment, marked as answer]
GOT IT!!! - if anyone is using the RLC-810A and having trouble with playback of clips recorded from the main stream @4k H265 add this input_args under the ffmpeg section for your camera. Crispy 4K clips now playing back on iOS.

```
ffmpeg:
      inputs:
        - path: rtsp://brandon:password@10.10.10.1:554//h264Preview_01_sub
          roles:
            - detect
            - rtmp
        - path: rtsp://brandon:pasword@10.10.10.1:554//h265Preview_01_main
          roles:
            - clips
      hwaccel_args:
        - -hwaccel
        - vaapi
        - -hwaccel_device
        - /dev/dri/renderD128
        - -hwaccel_output_format
        - yuv420p
      input_args: -tag:v hvc1
```

**Reply by blakeblackshear** [reply]
You may want to add those args in addition to the default args. Adding them this way replaces the defaults rather than adding to them.

**Reply by bmutari** [reply]
@blakeblackshear thanks. Question- is there a way to specify the `-tag:v hvc1` only on the main stream? I discovered that it is causing issues with the rtmp sub-stream as the sub-stream is H264 and the tag of hvc1 is only relevant on H265.

**Reply by gpete** [reply]
Yes, according to the [documentation](https://blakeblackshear.github.io/frigate/configuration/cameras) input_args can be specified for a specific stream. Nice find by the way!

**Reply by bmutari** [reply]
perfect! - just moved the input_args to underneath the main stream path and all looks to be working as expect. Thanks everyone for the help, and much thanks to @blakeblackshear Frigate is amazing!

**Reply by Red-ington** [reply]
Can you show your new config. I have a similar issue, also have a H265 C800 I91BM camera.

**Reply by bojangles9** [reply]
Trying to adapt this for a 820a. Can you please post the full config, and which part is actually fixing it ?
